### PR TITLE
fix: Remove`time_range_endpoints` from query context object

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

When running reports/alerts charts are getting the following 400 error, due deprecating time_range_endpoints field.
```
{"message":"Request is incorrect: {'queries': {0: {'extras': {'time_range_endpoints': ['Unknown field.']}}}}"}
```

To fix this we'll be running a migration to remove all the `time_range_endpoints` keys from the `slice.query_context`. We are only updating the query_context that are valid json

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [X] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [X] Confirm DB migration upgrade and downgrade tested
  - [X] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

```
Benchmark Results:
Current: 0.42 s
10+: 0.45 s
100+: 0.34 s
1000+: 0.33 s
Cleaning up DB
```

Downgrade:
```
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade 2ed890b36b94 -> 58df9d617f14, rm_time_range_endpoints_from_qc
```

Upgrade:
```
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 58df9d617f14 -> 2ed890b36b94, rm_time_range_endpoints_from_qc
```

### CODE REVIEW

File: superset/migrations/versions/2ed890b36b94_rm_time_range_endpoints_from_qc.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+from superset import db
+
+Base = declarative_base()
+
+
+class MyModel(Base):
+    __tablename__ = "MyModel"
+    id = sa.Column(sa.Integer, primary_key=True)
+    queries = sa.Column(sa.Text)
```
Comment (zhaoyongjie) [inline superset/migrations/versions/2ed890b36b94_rm_time_range_endpoints_from_qc.py:44]: Use readable model name.

```suggestion
class Slice(Base):
    __tablename__ = "slices"
    id = sa.Column(sa.Integer, primary_key=True)
    queries = sa.Column(sa.Text)
```

File: superset/migrations/versions/2ed890b36b94_rm_time_range_endpoints_from_qc.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    bind = op.get_bind()
+    session = db.Session(bind=bind)
+
+    for m in session.query(MyModel):
+        queries = json.loads(m.queries)
+        for q in queries:
+            extras = q.get("extras")
+            if extras:
+                extras.pop("time_range_endpoints", None)
+        m.queries = json.dumps(queries)
```
Comment (zhaoyongjie) [inline superset/migrations/versions/2ed890b36b94_rm_time_range_endpoints_from_qc.py:57]:
```suggestion
   for slc in session.query(Slice):
        queries = json.loads(slc.queries)
        for query in queries:
            query.get("extras", {}).pop("time_range_endpoints", None)
        m.queries = json.dumps(queries)
```

**Review by zhaoyongjie — COMMENTED** [review]
add some codes suggestion.

**Review by eschutho — APPROVED** [review]
Thanks @hughhhh. LGTM.

**Comment by eschutho** [comment]
@hughhhh before merging, I would recommend also running through the [db migration benchmark script](https://github.com/apache/superset/pull/13561) and checkboxes in the PR description and add the results of the benchmark to the description.

**Comment by eschutho** [comment]
This PR is a follow up to https://github.com/apache/superset/pull/18936

**Comment by yousoph** [comment]
Before this fix, users were getting an error when requesting a CSV or text report. PNG reports were working as expected 

Repro steps: 
1. Create a chart of a Table or Pivot Table
2. Add a time range filter onto your chart 
3. Create a report of your chart 

Current behavior: 
Report owner will receive an email with a 400 error instead of the report

File: superset/migrations/versions/2ed890b36b94_rm_time_range_endpoints_from_qc.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    for slc in session.query(Slice):
+        if slc.query_context:
+            try:
+                query_context = json.loads(slc.query_context)
+                queries = query_context.get("queries")
+                for query in queries:
+                    query.get("extras", {}).pop("time_range_endpoints", None)
+                slc.queries = json.dumps(queries)
+            except json.decoder.JSONDecodeError:
+                pass
```
Comment (betodealmeida) [inline superset/migrations/versions/2ed890b36b94_rm_time_range_endpoints_from_qc.py:59]: In general it's better to keep the try/except block as small as needed:

```suggestion
            try:
                query_context = json.loads(slc.query_context)
            except json.decoder.JSONDecodeError:
                continue
            queries = query_context.get("queries")
            for query in queries:
                query.get("extras", {}).pop("time_range_endpoints", None)
            slc.queries = json.dumps(queries)
```

**Review by betodealmeida — APPROVED** [review]
LGTM, just a suggestion on the try/except.

**Comment by sadpandajoe** [comment]
🏷️  preset:2022.11

**Comment by sadpandajoe** [comment]
🏷️ preset:2022.13

**Pull request merged**